Golem/Yagna/ProcessFactory.cs

```diff
@@ -60,77 +60,111 @@ public override void WriteLine()
 
     public class ProcessFactory
     {
-        public static Command CreateProcess<OUT_WRITER, ERR_WRITER>(string executable, IEnumerable<object>? args, Dictionary<string, string> env, OUT_WRITER stdOut, ERR_WRITER errOut)
-        where OUT_WRITER : TextWriter
-        where ERR_WRITER : TextWriter
+        public static Process StartProcess(string executable, IEnumerable<object> args, Dictionary<string, string> env, bool redirectOutput = false)
         {
-            var argList = args?.ToList();
-            argList?.RemoveAll(s => string.IsNullOrWhiteSpace((string?)s));
-            var executablePath = Path.GetFullPath(executable);
-            var workDir = Directory.GetParent(executablePath)?.ToString() ?? "";
-
-            return Command
-                .Run(executablePath, argList, options => updateOptions(options, workDir, env))
-                .RedirectTo(stdOut)
-                .RedirectStandardErrorTo(errOut);
+
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = executable,
+                RedirectStandardOutput = redirectOutput,
+                RedirectStandardError = redirectOutput,
+                CreateNoWindow = true,
```

Comment:
When `RedirectStandardOutput`/`Error` is set to `false` the process will not inherit console from parent process (in case of GUI app it would inherit none), and instead it will get a new console (so it will be able to receive CTRL-C signals).
`CreateNoWindow` makes console to not pop up.

Tradeoff of setting `RedirectStandardOutput`/`Error` to `false` is that we no longer can capture its output.
Not everything is lost, because logs are stored in `$HOME\AppData\Local\GamerHashAI\modules\golem_0.1.5\golem-data\provider`/`yagna`.
We set `RedirectStandardOutput` to `true` only when we run commands like `yagna id list` (in case of which we want to capture a stdout and we do not need to signal-kill these processes).
The problem is when e.g. process `args` are wrong, or path to file is incorrect, the error output will be printed to console, hence lost.

A solution not requiring setting `RedirectStandardOutput` to `true` could be starting instead of `yagna.exe`/`ya-provider.exe` a `cmd.exe` or `sh`, with `yagna.exe`/`ya-provider.exe` + their args as the `cmd`/`sh` args with ` > DATA_DIR/console.log  2>&1` as the last arg to just redirect console output to file.
The problem would be rolling these console logs (timestamp in filename for now? overwriting?), and disabling console logs in Rust logger (to not duplicate logs (but we could live with it for a while)), so only relevant console output would appear there (like stderr).

Reply:
A solution not requiring setting `RedirectStandardOutput` to `true` could be starting instead of `yagna.exe`/`ya-provider.exe` a `cmd.exe` or `sh`, with `yagna.exe`/`ya-provider.exe` + their args as the `cmd`/`sh` args with ` > DATA_DIR/console.log  2>&1` as the last arg to just redirect console output to file.
The problem would be rolling these console logs (timestamp in filename for now? overwriting?), and disabling console logs in Rust logger (to not duplicate logs (but we could live with it for a while)), so only relevant console output would appear there (like stderr).